### PR TITLE
Adopting lightweight generics

### PR DESCRIPTION
**Pull request by Coeur**

### CODE REVIEW

**Comment by ckerr** [comment]
@DevilDimon

**Comment by ckerr** [comment]
CI failure is unrelated to the PR

File: macosx/AboutWindowController.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // It may be used under the MIT (SPDX: MIT) license.
 // License text can be found in the licenses/ folder.
 
-#import <Cocoa/Cocoa.h>
+#import <AppKit/AppKit.h>
```
Comment (DevilDimon) [inline macosx/AboutWindowController.h:5]: Seems to be unrelated to the title of the PR?

File: macosx/BlocklistDownloader.mm [hunk 2]
```diff
@@ -15,7 +15,7 @@ @interface BlocklistDownloader ()
 @property(nonatomic) blocklistDownloadState fState;
 
 - (void)startDownload;
-- (void)decompressFrom:(NSURL*)file to:(NSURL*)destination error:(NSError**)error;
+- (BOOL)decompressFrom:(NSURL*)file to:(NSURL*)destination error:(NSError**)error;
```
Comment (DevilDimon) [inline macosx/BlocklistDownloader.mm:18]: Also seems to be unrelated to the title of the PR

File: macosx/VDKQueue/VDKQueue.m [hunk 3]
```diff
@@ -201,8 +201,6 @@ - (VDKQueuePathEntry *)    addPathToQueue:(NSString *)path notifyingAbout:(u_int
         
         return pathEntry;
     }
-    
-    return nil;
```
Comment (DevilDimon) [inline macosx/VDKQueue/VDKQueue.m:205]: This is third-party so we decided to keep it as is (note that it still uses ivars despite the fact that I removed them from the rest of the project earlier).

**Review by DevilDimon — APPROVED** [review]

File: macosx/AboutWindowController.h (hunk 1 above)
Comment (Coeur) [reply macosx/AboutWindowController.h]: Correct. Reverting.

File: macosx/VDKQueue/VDKQueue.m (hunk 3 above)
Comment (Coeur) [reply macosx/VDKQueue/VDKQueue.m]: OK, reverting.

File: macosx/BlocklistDownloader.mm (hunk 2 above)
Comment (Coeur) [reply macosx/BlocklistDownloader.mm:18]: This fixes a warning:
>Method accepting NSError** should have a non-void return value to indicate whether or not an error occurred

I think it's fine in this PR. Let me know if you prefer that I move it to a separate PR, @DevilDimon.
